### PR TITLE
Add ungridded dimensions and vertical dimensions for History3G

### PR DESCRIPTION
**Pull request by darianboggs**
## Types of change(s)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist
- [ ] Tested this change with a run of GEOSgcm
- [x] Ran the Unit Tests (`make tests`)

## Description
Add ungridded dimensions and vertical dimensions to History3G

### CODE REVIEW

**Comment by darianboggs** [comment]
Changed target to `release/MAPL-v3`

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added GitHub Action to generate MAPL3 Ford Docs
 - Added capability for HistoryCollectionGridComp to extract field names from expressions
 - Added ability for HistoryCollectionGridComp to extract multiple field names from expressions
+- Added vertical and ungridded dimensions to output for History3G
```
Comment (tclune) [inline CHANGELOG.md:38]:
```suggestion
- Added vertical and ungridded dimensions metadata to netCDF output for History3G
```

Comment (tclune) [reply CHANGELOG.md:38]: Please check if my wording is accurate.

**Comment by bena-nasa** [comment]
@darianboggs 
I tried running one of the test cases, but actually adding an ungridded dimension to one of the fields created in the leaf component. It failed saying that it could not make the connection in the generic layer. I suspect that we need modify how the import fields in the HistoryCollection component are defined. I'll take a look

**Comment by bena-nasa** [comment]
I managed to implement the ungridded dimension mirroring. But when I now try write fields with ungridded dimensions, the extent of the dimension put in the file is right, but the name is "NONE", not what is should be so not where this is getting lost.


Also @darianboggs it does not seem to be making a coordinate variable for the ungridded dimension

**Comment by bena-nasa** [comment]
@darianboggs can we get this merged? It seems to be working, the error I reported above are coming from the generic layer due to incomplete parsing of the ungridded dimensions, what have done seems to be doing what it is supposed to. I need to change the constructor for ungridded dims so better if this is merged first, then I'll make my PR of cleanup of the ungridded dimensions.

**Pull request merged**